the_brill_bridge: old DOIs resolve to .xml (202) instead of HTML article pages

## Problem

4 tests in `TestBrillDance` are failing because old Brill DOIs (10.1163/\*) now resolve to `.xml` endpoints at brill.com that return HTTP 202, not HTML article pages with `citation_pdf_url` meta tags.

```
FAILED tests/findit/test_brill.py::TestBrillDance::test_brill_bridge_url_construction_early_sci_med
FAILED tests/findit/test_brill.py::TestBrillDance::test_brill_bridge_url_construction_early_sci_med_alt
FAILED tests/findit/test_brill.py::TestBrillDance::test_brill_bridge_url_construction_phronesis
FAILED tests/findit/test_brill.py::TestBrillDance::test_brill_bridge_url_construction_toung_pao
```

## Example

DOI `10.1163/15733823-00202p03` resolves to:
```
https://brill.com/view/journals/esm/20/2/article-p150_3.xml  (HTTP 202, 1991 bytes)
```

The `.xml` response does not contain an HTML `citation_pdf_url` meta tag, so `the_brill_bridge` raises `NoPDFLink('MISSING: No PDF URL found via citation_pdf_url meta tag extraction')`.

## Affected PMIDs / DOIs

| PMID | Journal | DOI |
|------|---------|-----|
| 26415349 | Early Sci Med | 10.1163/15733823-00202p03 |
| 11873782 | Early Sci Med | 10.1163/157338201x00154 |
| 11618220 | Toung Pao | 10.1163/156853287x00032 |
| 11636720 | Phronesis (Barc) | 10.1163/156852873x00014 |

## Investigation needed

- Fetch the `.xml` response for one of these DOIs and determine if it contains a PDF link in XML format
- Check if there's an alternate HTML URL pattern for these older Brill articles (e.g. `.htm` instead of `.xml`)
- Update `the_brill_bridge` to handle the XML response format or construct the PDF URL differently

--- a/metapub/findit/dances/brill.py
+++ b/metapub/findit/dances/brill.py
@@ -1,14 +1,15 @@
 """
-Brill dance function - evidence-driven rewrite.
+Brill dance function.
 
-Evidence-driven implementation based on HTML samples analysis showing
-perfect citation_pdf_url meta tags in all samples.
+PDF URL construction strategy:
+- Resolve DOI via dx.doi.org to get the canonical article URL
+- Derive PDF URL by replacing /view/journals/ with /downloadpdf/journals/
+- This avoids scraping the article page entirely (which is behind AWS WAF)
+- Falls back to HTML citation_pdf_url meta tag parsing if URL pattern doesn't match
 
-Pattern discovered from HTML samples:
-- citation_pdf_url meta tags: 100% reliable
-- URL pattern: https://brill.com/downloadpdf/view/journals/{journal}/{volume}/{issue}/article-p{page}_{id}.pdf
-- DOI patterns: Multiple prefixes (10.1163, 10.1007, 10.1016, etc.) - no restriction needed
-- 403 Forbidden expected for subscription content but URLs are correctly constructed
+URL pattern (consistent across all Brill journals):
+  Article: https://brill.com/view/journals/{j}/{vol}/{iss}/article-p{page}_{id}.xml
+  PDF:     https://brill.com/downloadpdf/journals/{j}/{vol}/{iss}/article-p{page}_{id}.xml
 """
 
 import re
@@ -16,47 +17,70 @@ import requests
 from .generic import the_doi_2step, verify_pdf_url, unified_uri_get
 from ...exceptions import NoPDFLink, AccessDenied
 
+BRILL_VIEW_PATH = '/view/journals/'
+BRILL_PDF_PATH = '/downloadpdf/journals/'
+
 
 def the_brill_bridge(pma, verify=True, request_timeout=10, max_redirects=3):
     """
-    Brill dance function - evidence-driven meta tag extraction.
-    
-    Based on evidence analysis showing perfect citation_pdf_url meta tags
-    in all HTML samples. Simple and reliable approach.
-    
+    Brill dance function.
+
+    Constructs the PDF URL from the DOI-resolved article URL without scraping
+    the article page (which is behind AWS WAF bot protection).
+
     :param pma: PubMedArticle object
     :param verify: bool [default: True] - Verify PDF URL accessibility
     :return: PDF URL string
-    :raises: NoPDFLink if DOI missing or meta tag extraction fails
+    :raises: NoPDFLink if DOI missing or URL construction fails
+    :raises: AccessDenied if Brill WAF blocks verification
     """
     if not pma.doi:
         raise NoPDFLink('MISSING: DOI required for Brill articles')
-    
-    # Resolve DOI to get article HTML page
+
     article_url = the_doi_2step(pma.doi)
-    
+
+    if not article_url:
+        raise NoPDFLink('MISSING: Could not resolve Brill DOI to article URL')
+
+    # Derive PDF URL directly from article URL — no need to fetch the article page
+    if BRILL_VIEW_PATH in article_url:
+        pdf_url = article_url.replace(BRILL_VIEW_PATH, BRILL_PDF_PATH)
+    else:
+        # Fallback: fetch the page and look for citation_pdf_url meta tag
+        pdf_url = _extract_pdf_url_from_page(article_url, request_timeout, max_redirects)
+
+    if verify:
+        try:
+            verify_pdf_url(pdf_url, 'Brill', request_timeout=request_timeout, max_redirects=max_redirects)
+        except Exception:
+            # Fetch the page to check if it's a WAF block, for a clearer error
+            try:
+                response = unified_uri_get(article_url, timeout=request_timeout, max_redirects=max_redirects)
+                if 'awswaf' in response.text or 'challenge.js' in response.text:
+                    raise AccessDenied('DENIED: Brill is protected by AWS WAF bot detection (requires browser)')
+            except (requests.exceptions.ConnectionError, requests.exceptions.Timeout):
+                pass
+            raise
+
+    return pdf_url
+
+
+def _extract_pdf_url_from_page(article_url, request_timeout, max_redirects):
+    """Fallback: fetch article page and extract citation_pdf_url meta tag."""
     try:
         response = unified_uri_get(article_url, timeout=request_timeout, max_redirects=max_redirects)
     except requests.exceptions.ConnectionError as e:
         raise NoPDFLink(f'TXERROR: Network error accessing Brill article: {e}')
-    
+
     if response.status_code not in (200, 202, 301, 302, 307):
         raise NoPDFLink(f'TXERROR: Could not access Brill article page (HTTP {response.status_code})')
 
-    # Detect AWS WAF JavaScript challenge (returned as 202 with challenge page)
     if 'awswaf' in response.text or 'challenge.js' in response.text:
         raise AccessDenied('DENIED: Brill is protected by AWS WAF bot detection (requires browser)')
 
-    # Extract citation_pdf_url meta tag (evidence-based approach)
     pdf_match = re.search(r'<meta[^>]*name=["\']citation_pdf_url["\'][^>]*content=["\']([^"\']+)["\']',
-                         response.text, re.IGNORECASE)
-
+                          response.text, re.IGNORECASE)
     if not pdf_match:
         raise NoPDFLink('MISSING: No PDF URL found via citation_pdf_url meta tag extraction')
-    
-    pdf_url = pdf_match.group(1)
-    
-    if verify:
-        verify_pdf_url(pdf_url, 'Brill', request_timeout=request_timeout, max_redirects=max_redirects)
-    
-    return pdf_url
+
+    return pdf_match.group(1)

--- a/tests/findit/test_brill.py
+++ b/tests/findit/test_brill.py
@@ -19,70 +19,69 @@ class TestBrillDance(BaseDanceTest):
         super().setUp()
         self.fetch = PubMedFetcher()
 
-    def test_brill_bridge_waf_early_sci_med(self):
-        """Test 1: Brill AWS WAF detection (Early Sci Med).
+    def test_brill_bridge_url_construction_early_sci_med(self):
+        """Test 1: PDF URL construction without verification (Early Sci Med).
 
-        PMID: 26415349 (Early Sci Med)
-        Brill serves AWS WAF JavaScript challenge pages (202) for these articles.
-        Expected: AccessDenied with clear WAF message, not confusing MISSING error.
+        PMID: 26415349 - DOI resolves to brill.com/view/journals/esm/...
+        Expected: returns downloadpdf URL without hitting the WAF-blocked article page.
         """
         pma = load_pmid_xml('26415349')
 
         assert pma.journal == 'Early Sci Med'
         assert pma.doi == '10.1163/15733823-00202p03'
 
-        with pytest.raises(AccessDenied) as exc_info:
-            the_brill_bridge(pma, verify=False)
-        assert 'WAF' in str(exc_info.value) or 'bot' in str(exc_info.value).lower()
-        print(f"Test 1 - Correctly identified WAF block: {exc_info.value}")
+        url = the_brill_bridge(pma, verify=False)
+        assert url is not None
+        assert 'brill.com/downloadpdf/journals/' in url
+        print(f"Test 1 - PDF URL: {url}")
 
-    def test_brill_bridge_waf_early_sci_med_alt(self):
-        """Test 2: Brill AWS WAF detection (Early Sci Med alt).
+    def test_brill_bridge_url_construction_early_sci_med_alt(self):
+        """Test 2: PDF URL construction without verification (Early Sci Med alt).
 
-        PMID: 11873782 (Early Sci Med)
-        Expected: AccessDenied with clear WAF message.
+        PMID: 11873782
+        Expected: returns downloadpdf URL.
         """
         pma = load_pmid_xml('11873782')
 
         assert pma.journal == 'Early Sci Med'
         assert pma.doi == '10.1163/157338201x00154'
 
-        with pytest.raises(AccessDenied) as exc_info:
-            the_brill_bridge(pma, verify=False)
-        assert 'WAF' in str(exc_info.value) or 'bot' in str(exc_info.value).lower()
-        print(f"Test 2 - Correctly identified WAF block: {exc_info.value}")
+        url = the_brill_bridge(pma, verify=False)
+        assert url is not None
+        assert 'brill.com/downloadpdf/journals/' in url
+        print(f"Test 2 - PDF URL: {url}")
 
-    def test_brill_bridge_waf_toung_pao(self):
-        """Test 3: Brill AWS WAF detection (Toung Pao).
+    def test_brill_bridge_url_construction_toung_pao(self):
+        """Test 3: PDF URL construction without verification (Toung Pao).
 
-        PMID: 11618220 (Toung Pao)
-        Expected: AccessDenied with clear WAF message.
+        PMID: 11618220
+        Expected: returns downloadpdf URL.
         """
         pma = load_pmid_xml('11618220')
 
         assert pma.journal == 'Toung Pao'
         assert pma.doi == '10.1163/156853287x00032'
 
-        with pytest.raises(AccessDenied) as exc_info:
-            the_brill_bridge(pma, verify=False)
-        assert 'WAF' in str(exc_info.value) or 'bot' in str(exc_info.value).lower()
-        print(f"Test 3 - Correctly identified WAF block: {exc_info.value}")
+        url = the_brill_bridge(pma, verify=False)
+        assert url is not None
+        assert 'brill.com/downloadpdf/journals/' in url
+        print(f"Test 3 - PDF URL: {url}")
 
-    def test_brill_bridge_waf_phronesis(self):
-        """Test 4: Brill AWS WAF detection (Phronesis).
+    def test_brill_bridge_url_construction_phronesis(self):
+        """Test 4: PDF URL construction without verification (Phronesis).
 
-        PMID: 11636720 (Phronesis)
-        Expected: AccessDenied with clear WAF message.
+        PMID: 11636720
+        Expected: returns downloadpdf URL.
         """
         pma = load_pmid_xml('11636720')
 
         assert pma.journal == 'Phronesis (Barc)'
         assert pma.doi == '10.1163/156852873x00014'
 
-        with pytest.raises(AccessDenied) as exc_info:
-            the_brill_bridge(pma, verify=False)
-        assert 'WAF' in str(exc_info.value) or 'bot' in str(exc_info.value).lower()
-        print(f"Test 4 - Correctly identified WAF block: {exc_info.value}")
+        url = the_brill_bridge(pma, verify=False)
+        assert url is not None
+        assert 'brill.com/downloadpdf/journals/' in url
+        print(f"Test 4 - PDF URL: {url}")
 
     # Test removed: Multiple tests - successful access, paywall detection, access forbidden, network error - functionality now handled by verify_pdf_url
 
@@ -186,27 +185,21 @@ class TestBrillXMLFixtures:
             pma = load_pmid_xml(pmid)
             assert pma.doi.startswith(doi_prefix), f"PMID {pmid} XML fixture has unexpected DOI: {pma.doi}"
 
-    @patch('metapub.findit.dances.brill.unified_uri_get')
     @patch('metapub.findit.dances.brill.the_doi_2step')
-    def test_brill_template_flexibility(self, mock_doi_2step, mock_uri_get):
-        """Test template flexibility for Brill URL patterns."""
+    def test_brill_template_flexibility(self, mock_doi_2step):
+        """Test PDF URL derivation from article URL pattern.
+
+        The function derives the PDF URL directly from the DOI-resolved article
+        URL without fetching the article page (which is behind AWS WAF).
+        Pattern: /view/journals/ -> /downloadpdf/journals/
+        """
         pma = load_pmid_xml('26415349')  # Early Sci Med
 
-        # Mock successful response with citation_pdf_url meta tag
-        mock_response = Mock()
-        mock_response.status_code = 200
-        expected_pdf_url = 'https://brill.com/downloadpdf/view/journals/early-sci-med-test.pdf'
-        mock_response.text = f'<html><head><meta name="citation_pdf_url" content="{expected_pdf_url}" /></head></html>'
-        mock_uri_get.return_value = mock_response
+        article_url = 'https://brill.com/view/journals/esm/20/2/article-p150_3.xml'
+        mock_doi_2step.return_value = article_url
 
-        # Mock DOI resolution
-        expected_article_url = 'https://brill.com/view/journals/early-sci-med-test'
-        mock_doi_2step.return_value = expected_article_url
-
-        # Test URL construction
         result = the_brill_bridge(pma, verify=False)
 
-        # Should follow Brill URL pattern
-        assert result == expected_pdf_url
-        assert 'brill.com' in result
+        assert result == 'https://brill.com/downloadpdf/journals/esm/20/2/article-p150_3.xml'
+        assert 'downloadpdf/journals' in result
         mock_doi_2step.assert_called_with(pma.doi)    # Test removed: test_brill_historical_journals_coverage - functionality now handled by verify_pdf_url

